src/store/slices/prevEmployees.slice.ts

```diff
@@ -0,0 +1,55 @@
+import { Action, Employee } from "../../models";
+import { Dispatch } from "../store";
+
+export type PrevEmployeesState = Map<String, Partial<Employee>>;
+
+const SET_PREV_EMPLOYEE = "SET_PREV_EMPLOYEE";
+const DELETE_PREV_EMPLOYEE = "DELETE_PREV_EMPLOYEE";
+
+const prevEmployeesReducer = (
+  state: PrevEmployeesState = new Map(),
+  action: Action
+) => {
+  switch (action.type) {
+    case SET_PREV_EMPLOYEE: {
+      const newState = new Map(state);
+      newState.set(action.payload.id, action.payload.employee);
+      return newState;
+    }
+    case DELETE_PREV_EMPLOYEE: {
+      const newState = new Map(state);
+      if (newState.get(action.payload.id) === action.payload.employee) {
+        newState.delete(action.payload.id);
+      }
+      return newState;
+    }
+    default: {
+      return state;
+    }
+  }
+};
```

Comment:
good to move the slices to respective folders than putting everything together, so it will be easy for new developers to read and use